lib/customer-deposit-wallet/src/Cardano/Wallet/Deposit/Pure.hs

```diff
@@ -80,10 +80,16 @@ import Cardano.Wallet.Deposit.Pure.API.TxHistory
     ( ByCustomer
     , ByTime
     , DownTime
+    , ResolveSlot
```

Comment:
```suggestion
```

In the context of `rollForwardOne`, I think that the type `ResolveSlot` no longer sits at the right abstraction level when imported from `Cardano.Wallet.Deposit.Pure.API.TxHistory`. 🤔 

Could you add a type in the following module

```hs
module Cardano.Wallet.Deposit.Time (… LookupTimeFromSlot …) where
…
type LookupTimeFromSlot = Slot -> Maybe (WithOrigin UTCTime)
```

instead and use that one in place of `ResolveSlot`? It's almost the same type, except

* More descriptive name `LookupTimeFromSlot`
* Removal of `Down`, as the sorting order is an interface detail for `TxHistory`